python/nvtx/_lib/lib.pyx

```diff
@@ -14,36 +14,32 @@ from nvtx.utils.cached import CachedInstanceMeta
 cpdef bytes _to_bytes(object s):
     return s if isinstance(s, bytes) else s.encode()
 
-
 def initialize():
     nvtxInitialize(NULL)
 
 
 cdef class EventAttributes:
 
     def __init__(self, object message=None, color=None, category=None):
-        self._color = color
-        self.message = message
         self.c_obj = nvtxEventAttributes_t(0)
         self.c_obj.version = NVTX_VERSION
         self.c_obj.size = NVTX_EVENT_ATTRIB_STRUCT_SIZE
         self.c_obj.colorType = NVTX_COLOR_ARGB
-        self.c_obj.color = color_to_hex(self._color)
-        self.c_obj.messageType = NVTX_MESSAGE_TYPE_ASCII
-        self.c_obj.message.ascii = self._message
-        if category is not None:
-            self.c_obj.category = category
+        self.c_obj.messageType = NVTX_MESSAGE_TYPE_REGISTERED
+
+        self.message = message
+        self.color = color
+        self.category = category
+
 
     @property
     def message(self):
-        return self._message.decode()
+        return self._message.string
 
     @message.setter
     def message(self, object value):
-        if value is None:
-            value = b""
-        self._message = _to_bytes(value)
-        self.c_obj.message.ascii = self._message
+        self._message = value
+        self.c_obj.message.registered = (<StringHandle> self._message.handle).c_obj
```

Comment:
Since the `getter` returns a `str`, perhaps we should make the `setter` as well accept a `str`?

Reply:
The issue is that the RegisteredString constructor requires a domain. An alternative would be to replace the setter with a function.

Reply:
Ah, then for symmetry, perhaps the `getter` should return the `RegisteredString`.

Reply:
Getter now returns object.